lib/libc/Kconfig

```diff
@@ -36,7 +36,6 @@ config PICOLIBC_SUPPORTED
 	bool
 	depends on !NATIVE_APPLICATION
 	depends on ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y") || (NATIVE_LIBRARY)
-	depends on !(CPP && ("$(TOOLCHAIN_HAS_PICOLIBC)" = "y"))
```

Comment:
This piece looks correct -- you can use C++ with the picolibc module as long as you don't use a pre-built standard C++ library.

Reply:
Yeah, this works for me. We may need to make it more clear that when you use `PICOLIBC_USE_MODULE` that it doesn't contain C++ library.